### PR TITLE
vimPlugins.nvim-treesitter.grammarToPlugin: init

### PR DESCRIPTION
**Pull request by figsoda**
###### Description of changes

second attempt at #207908, see https://github.com/NixOS/nixpkgs/pull/225152#issuecomment-1505341431

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/editors/vim/plugins/nvim-treesitter/overrides.nix [hunk 1]
```diff
@@ -27,6 +27,25 @@ let
...
+        # added in buildGrammar
+        (lib.removeSuffix "-grammar")
+
+        # grammars from tree-sitter.builtGrammars
+        (lib.removePrefix "tree-sitter-")
+        (lib.replaceStrings [ "-" ] [ "_" ])
+      ];
+    in
+
+    runCommand "nvim-treesitter-${name}-grammar" { } ''
```
Comment (teto) [inline pkgs/applications/editors/vim/plugins/nvim-treesitter/overrides.nix:44]: I would prefer `runCommand "nvim-treesitter-grammar-${name}" { } ''` but if that breaks some coherency later on let's forget about it

File: pkgs/applications/editors/vim/plugins/nvim-treesitter/overrides.nix [hunk 2]
```diff
@@ -27,6 +27,25 @@ let
...
+      mkdir -p $out/parser
+      ln -s ${grammar}/parser $out/parser/${name}.so
```
Comment (teto) [inline pkgs/applications/editors/vim/plugins/nvim-treesitter/overrides.nix:46]: the .so will work on linux only. wouldn't something like `${stdenv.hostPlatform.extensions.sharedLibrary}` be better ?

**Comment by teto** [comment]
thnaks, exactly what I was looking for

File: pkgs/applications/editors/vim/plugins/nvim-treesitter/overrides.nix (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/applications/editors/vim/plugins/nvim-treesitter/overrides.nix:46]: as a suggestion 
```suggestion
      ln -s ${grammar}/parser $out/parser/${name}.${stdenv.hostPlatform.extensions.sharedLibrary}
```

File: pkgs/applications/editors/vim/plugins/nvim-treesitter/overrides.nix (hunk 1 above)
Comment (figsoda) [reply pkgs/applications/editors/vim/plugins/nvim-treesitter/overrides.nix]: done

**Pull request merged**